src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -863,6 +863,17 @@ struct ONNXElementwiseUnaryOpLowering : public ConversionPattern {
     auto loc = ONNXLoc<ElementwiseUnaryOp>(op);
     auto X = operands[0];
 
+    // If type is scalar, there need not allocate a buffer. Just call scalar
+    // computation and return the result.
+    // This is efficient when elementwise ops are used as activations for ops
+    // like LSTM/GRU/RNN.
+    if (!X.getType().isa<TensorType>() && !X.getType().isa<MemRefType>()) {
```

Comment:
I am wondering whether this could be written as: 
```
bool isScalar = !X.getType().isa<ShapedType>();
if (!isScalar) {
```

Reply:
ShapedType covers VectorType. That's why I didn't use it. I updated the comment to mention vector type. Thanks!